powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/TokenAndEncryptStep.java

```diff
@@ -0,0 +1,172 @@
+        );
+    }
+
+    @Override
+    protected ParameterizedTypeReference<EciesEncryptedResponse> getResponseTypeReference() {
+        return PowerAuthConst.RESPONSE_TYPE_REFERENCE_V3;
+    }
+
+    @Override
+    public StepContext<TokenAndEncryptStepModel, EciesEncryptedResponse> prepareStepContext(StepLogger stepLogger, Map<String, Object> context) throws Exception {
+        TokenAndEncryptStepModel model = new TokenAndEncryptStepModel();
+        model.fromMap(context);
+
+        RequestContext requestContext = RequestContext.builder()
+                .signatureHttpMethod(model.getHttpMethod())
+                .uri(model.getUriString())
+                .build();
+
+        StepContext<TokenAndEncryptStepModel, EciesEncryptedResponse> stepContext =
+                buildStepContext(stepLogger, model, requestContext);
+
+        // Verify that HTTP method is set
+        if (model.getHttpMethod() == null) {
+            stepLogger.writeError("token-encrypt-error-http-method", "HTTP method not specified", "Specify HTTP method to use for sending request");
+            stepLogger.writeDoneFailed("token-encrypt-failed");
+            return null;
+        }
+
+        // Verify HTTP method, GET is not supported
+        if (HttpMethod.GET.name().equals(model.getHttpMethod().toUpperCase())) {
+            stepLogger.writeError("token-encrypt-error-http-method-invalid", "Token and Encrypt Request Failed", "Unsupported HTTP method: " + model.getHttpMethod().toUpperCase());
+            stepLogger.writeDoneFailed("token-encrypt-failed");
+            return null;
+        }
+
+        // Read data which needs to be encrypted
+        byte[] requestDataBytes = model.getData();
+        if (requestDataBytes == null || requestDataBytes.length == 0) {
+            requestDataBytes = new byte[0];
+            stepLogger.writeItem(
+                    "token-validate-warning-empty-data",
+                    "Empty data",
+                    "Data file was not found, signature will contain no data",
+                    "WARNING",
+                    null
+            );
+        }
+
+        stepLogger.writeItem(
+                getStep().id() + "-request-prepare",
+                "Preparing Request Data",
+                "Following data will be encrypted",
+                "OK",
+                requestDataBytes
+        );
+
+        requestContext.setRequestObject(requestDataBytes);
+        powerAuthHeaderFactory.getHeaderProvider(model).addHeader(stepContext);
+
+        // Encrypt the request
+        addEncryptedRequest(stepContext, model.getApplicationSecret(), EciesSharedInfo1.ACTIVATION_SCOPE_GENERIC, requestDataBytes);
+
+        return stepContext;
+    }
+
+    @Override
+    public void processResponse(StepContext<TokenAndEncryptStepModel, EciesEncryptedResponse> stepContext) throws Exception {
+        ResponseContext<EciesEncryptedResponse> responseContext = stepContext.getResponseContext();
```

Comment:
same 14 lines with the `SignAndEncryptStep#processResponse` method

Reply:
The lines also repeat in `EncryptStep`, I will prepare a utility class for this.